### PR TITLE
Hacktoberfest 2020: Conway's Game of Life

### PR DESCRIPTION
**Pull request by jenra-uwu**
### **Describe your change:**
Added Conway's Game of Life.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by poyea** [comment]
I see another Game of Life in this repo. Do you think we can move these 2 (yours and that one) into one folder? https://github.com/TheAlgorithms/Python/blob/master/other/game_of_life.py

File: cellular_automata/conways_game_of_life.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    [0, 0, 0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0, 0, 0],
+]
+
+# Define blinker example
+BLINKER = [[0, 1, 0], [0, 1, 0], [0, 1, 0]]
+
+
+def new_generation(cells: list[list[int]]) -> list[list[int]]:
```
Comment (poyea) [inline cellular_automata/conways_game_of_life.py:26]:
```suggestion
def new_generation(cells: List[List[int]]) -> List[List[int]]:
```

**Comment by jenra-uwu** [comment]
Ohp I didn't realise that existed! Yeah we should probably move them both into the cellular_automata folder right?

**Comment by cclauss** [comment]
`from typing import List`

Fixes...
```
cellular_automata/conways_game_of_life.py:26:27: F821 undefined name 'List'
cellular_automata/conways_game_of_life.py:26:32: F821 undefined name 'List'
cellular_automata/conways_game_of_life.py:26:47: F821 undefined name 'List'
cellular_automata/conways_game_of_life.py:26:52: F821 undefined name 'List'
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**